docs/core/extensions/snippets/logging/library-authors/NonDiExampleService.cs

```diff
@@ -0,0 +1,20 @@
+﻿using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+
+namespace Logging.LibraryAuthors;
+
+public class NonDiExampleService
+{
+    private readonly ILogger<NonDiExampleService> _logger;
+
+    public NonDiExampleService() =>
+        _logger = NullLoggerFactory.Instance.CreateLogger<NonDiExampleService>();
+
```

Comment:
We need another constructor overload that takes an ILoggerFactory right? We want to show developers that pattern of writing a library public API that consumes ILoggerFactory.

Reply:
I could be wrong, but I thought that we didn't want to show anything that could be confused with DI. I thought having a constructor accept an `ILoggerFactory` might be really confusing. The issue (and library) that spawned this entire article is planning on accessing logging functionality statically post-configuration.

Reply:
Its fine if we don't show it as a constructor, but we need to show some API that takes ILoggerFactory as an input, creates an ILogger from it, and then logs to it. This example only logs to NullLogger and never anything else. For example:

```C#
class MyLibrary
{
    internal static ILoggerFactory LoggerFactory { get; private set; } = NullLoggerFactory;

    public static void SetLoggerFactory(ILoggerFactory loggerFactory) { LoggerFactory = loggerFactory; }
}

class NonDiExampleService
{
    private readonly ILogger<NonDiExampleService> _logger;
    public NonDiExampleService() =>
        _logger = MyLibrary.LoggerFactory.CreateLogger<NonDiServiceExample>();
    public void DoSomething()
    {
        LogWork(_logger);
    }
    
    [LoggerMessage(Message="Doing work")]
    partial static void LogWork(ILogger logger);
}

Reply:
In terms of the library API design I don't think we have a lot of precedent to recommend exactly how they structure it, but I am imagining a static setter method, static property, or constructor parameter are the most likely techniques library authors would want to use.